src/core/qgsvectorfilewriter.cpp

```diff
@@ -650,7 +650,16 @@ void QgsVectorFileWriter::init( QString vectorFileName,
             break;
 
           case QVariant::Double:
-            ogrType = OFTReal;
+            if ( ( mOgrDriverName == QLatin1String( "GPKG" ) || mOgrDriverName == QLatin1String( "SQLite" ) ) &&
```

Comment:
we only need this for GPKG
```suggestion
            if ( mOgrDriverName == QLatin1String( "GPKG") &&
```

Reply:
@rouault , shouldn't that also apply to spatialite datasets?

Reply:
Ignore me :) thanks for the review.

Reply:
> shouldn't that also apply to spatialite datasets?

Potentially yes, but as QGIS doesn't expose the FID column of Spatialite datasets as an attribute column, the practical risk is much lower